Skip object locking tests on NAS gateway

```
~ minio gateway nas ~/test
Endpoint:  http://10.0.0.67:9000  http://172.17.0.1:9000  http://172.18.0.1:9000  http://127.0.0.1:9000                  
AccessKey: minio 
SecretKey: minio123 

Browser Access:
   http://10.0.0.67:9000  http://172.17.0.1:9000  http://172.18.0.1:9000  http://127.0.0.1:9000                  

Command-line Access: https://docs.min.io/docs/minio-client-quickstart-guide
   $ mc alias set mynas http://10.0.0.67:9000 minio minio123

Object API (Amazon S3 compatible):
   Go:         https://docs.min.io/docs/golang-client-quickstart-guide
   Java:       https://docs.min.io/docs/java-client-quickstart-guide
   Python:     https://docs.min.io/docs/python-client-quickstart-guide
   JavaScript: https://docs.min.io/docs/javascript-client-quickstart-guide
   .NET:       https://docs.min.io/docs/dotnet-client-quickstart-guide
```

```
~  ./gradlew runFunctionalTest -Pendpoint=http://localhost:9000 -PaccessKey=minio -PsecretKey=minio123
...
listenBucketNotification()
listObjects()
putObject()
<FAILED> putObject() [with retention]
error occurred
ErrorResponse(code = NoSuchBucket, message = The specified bucket does not exist, bucketName = minio-java-test-141qq7u, objectName = null, resource = /minio-java-test-141qq7u, requestId = 164BE11C0355127F, hostId = 6504773c-3b8b-4ecc-af50-bf3d360ba49f)
request={method=GET, url=http://localhost:9000/minio-java-test-141qq7u?location=, headers=Host: localhost:9000
Accept-Encoding: identity
User-Agent: MinIO (Linux; amd64) minio-java/8.0.2-DEV
Content-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20201129T044642Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20201129/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=404, headers=Accept-Ranges: bytes
Content-Length: 318
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Server: MinIO
Vary: Origin
X-Amz-Request-Id: 164BE11C0355127F
X-Xss-Protection: 1; mode=block
Date: Sun, 29 Nov 2020 04:46:42 GMT
}

        at io.minio.MinioClient.execute(MinioClient.java:775)
        at io.minio.MinioClient.getRegion(MinioClient.java:807)
        at io.minio.MinioClient.putObject(MinioClient.java:4534)
        at io.minio.MinioClient.putObject(MinioClient.java:2696)
        at io.minio.MinioClient.putObject(MinioClient.java:2813)
        at FunctionalTest.testPutObject(FunctionalTest.java:756)
        at FunctionalTest.putObject(FunctionalTest.java:929)
        at FunctionalTest.runObjectTests(FunctionalTest.java:3725)
        at FunctionalTest.runTests(FunctionalTest.java:3757)
        at FunctionalTest.main(FunctionalTest.java:3901)

> Task :functional:runFunctionalTest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':functional:runFunctionalTest'.
> Process 'command '/usr/lib/jvm/java-8-openjdk-amd64/bin/java'' finished with non-zero exit value 255

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.6.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 35s
4 actionable tasks: 1 executed, 3 up-to-date
```

This is from release branch 8.0.2 @balamurugana

--- a/functional/FunctionalTest.java
+++ b/functional/FunctionalTest.java
@@ -689,6 +689,13 @@ public class FunctionalTest {
       client.makeBucket(
           MakeBucketArgs.builder().bucket(bucketNameWithLock).objectLock(true).build());
     } catch (Exception e) {
+      if (e instanceof ErrorResponseException) {
+        if (((ErrorResponseException) e).errorResponse().code().equals("NotImplemented")) {
+          bucketNameWithLock = null;
+          return;
+        }
+      }
+
       handleException("makeBucket()", "[object lock]", startTime, e);
     }
   }
@@ -926,14 +933,16 @@ public class FunctionalTest {
             .build(),
         null);
 
-    testPutObject(
-        "[with retention]",
-        PutObjectArgs.builder().bucket(bucketNameWithLock).object(getRandomName()).stream(
-                new ContentInputStream(1 * KB), 1 * KB, -1)
-            .retention(
-                new Retention(RetentionMode.GOVERNANCE, ZonedDateTime.now(Time.UTC).plusDays(1)))
-            .build(),
-        null);
+    if (bucketNameWithLock != null) {
+      testPutObject(
+          "[with retention]",
+          PutObjectArgs.builder().bucket(bucketNameWithLock).object(getRandomName()).stream(
+                  new ContentInputStream(1 * KB), 1 * KB, -1)
+              .retention(
+                  new Retention(RetentionMode.GOVERNANCE, ZonedDateTime.now(Time.UTC).plusDays(1)))
+              .build(),
+          null);
+    }
 
     testThreadedPutObject();
 
@@ -2313,6 +2322,8 @@ public class FunctionalTest {
   }
 
   public static void enableObjectLegalHold() throws Exception {
+    if (bucketNameWithLock == null) return;
+
     String methodName = "enableObjectLegalHold()";
     if (!mintEnv) {
       System.out.println(methodName);
@@ -2351,6 +2362,8 @@ public class FunctionalTest {
   }
 
   public static void disableObjectLegalHold() throws Exception {
+    if (bucketNameWithLock == null) return;
+
     String methodName = "disableObjectLegalHold()";
     if (!mintEnv) {
       System.out.println(methodName);
@@ -2391,6 +2404,8 @@ public class FunctionalTest {
   }
 
   public static void isObjectLegalHoldEnabled() throws Exception {
+    if (bucketNameWithLock == null) return;
+
     String methodName = "isObjectLegalHoldEnabled()";
     if (!mintEnv) {
       System.out.println(methodName);
@@ -2534,6 +2549,8 @@ public class FunctionalTest {
   }
 
   public static void setObjectRetention() throws Exception {
+    if (bucketNameWithLock == null) return;
+
     String methodName = "setObjectRetention()";
     if (!mintEnv) {
       System.out.println(methodName);
@@ -2614,6 +2631,8 @@ public class FunctionalTest {
   }
 
   public static void getObjectRetention() throws Exception {
+    if (bucketNameWithLock == null) return;
+
     String methodName = "getObjectRetention()";
     if (!mintEnv) {
       System.out.println(methodName);

